File: BaekJoon_2004.py
# Baekjoon 2004
# https://www.acmicpc.net/problem/2004

n, m = map(int, input().split())

lst = [0, 0]
i = 2
j = 5
while i <= n:
    lst[0] += n // i - m // i - (n-m) // i
    lst[1] += n // j - m // j - (n - m) // j
    i *= 2
    j *= 5

print(min(lst))

File: BaekJoon_2004.java
// Baekjoon 2004
// https://www.acmicpc.net/problem/2004

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int num2 = 0;
        int num5 = 0;

        int n = sc.nextInt();
        int m = sc.nextInt();
        sc.close();

        for (long i=2L, j=5L; i<=n; i*=2L, j*=5L) {
            num2 += n/i - m/i - (n-m)/i;
            num5 += n/j - m/j - (n-m)/j;
        }
        System.out.println(Math.min(num2, num5));
    }
}

